### PR TITLE
TopNav: Make dashboard toolbar actions work in new top nav 

### PR DESCRIPTION
**Pull request by torkelo**
This was very hard, I struggled to find a solution I am happy with in the last 2 days. But I think this is a good enough solution for now. 

The thing that complicates this are
* Supporting both toolbars at the same time 
* Different look for default ToolbarButton in new design 
* Trying to think ahead of how we are to add the responsive behavior to the toolbar (where actions collapse into a menu when they don't fit, like google docs toolbar). 

PR Changes
* Feature flags added to theme so components can look different when feature toggle is enabled (grafana/ui doesn't not have access to runtime/config) 
* This required some changes to ThemeProvider so that the same theme instance is used by config.theme2 and react theme context  
* ToolbarButton automatically look different when topnav feature is enabled 
* Added a NavToolbarButtonSeparator to handle left aligning some actions and line separator (Likely a temporary solution) 
* DashNav pass all actions via AppChromeUpdate actions 

Things to fix in future PRs
* ToolbarButton is used outside the nav toolbar in Panel edit, we might have to do a panel edit redesign or find some new ToolbarButton variant "outside" 
* How to get the separator lines between the elements in TimeRange picker 
* Want to move the kiosk feature to be handled at the application level (so all pages support it, and so we don't show the kiosk button until the top search bar is hidden) 
* Progressive / responsive "collapse to menu" feature for navbar actions (thinking it might be possible to do with IntersectionObserver, but it will be challenging and complex) 
* Fixing sectionNav (You will notice in breadcrumbs it just says Grafana in the breadcrumbs)

As we address the above we might have to tweak this solution or make interface changes to AppChromeUpdate actions param. Like adding a wrapper around the actions with some descriptor to help the responsive behavior handling.

### CODE REVIEW

**Review by joshhunt — COMMENTED** [review]
Nice stuff. Toolbar buttons aren't quite there yet (mainly the segmented refresh picker), but can be iterated on!!!!!

File: public/app/core/components/AppChrome/AppChromeUpdate.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { appChromeService } from './AppChromeService';
 
 export interface AppChromeUpdateProps {
   pageNav?: NavModelItem;
-  actions?: React.ReactNode;
+  actions?: React.ReactNode[];
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/core/components/AppChrome/AppChromeUpdate.tsx:9]: @joshhunt yes, the reason I went with array is thinking ahead of how to deal with progressive responsive behavior of collapsing actions that do not fit into a menu. To do that we need to iterate each action and render items that do not fit in the menu so each action will need some isolation. 

I can see this array turning into a actions?: NavBarAction[] kind of wrapper object in the future, but all depends on how the responsive behavior will be implemented .

Check google docs toolbar when the view width narrows for what I am thinking of

File: public/app/features/dashboard/components/DashNav/DashNav.tsx [hunk 2]
```diff
@@ -279,7 +280,11 @@ export const DashNav = React.memo<Props>((props) => {
     return (
       <AppChromeUpdate
         pageNav={{ text: title }}
-        actions={<ToolbarButton onClick={onOpenSettings} icon="cog"></ToolbarButton>}
+        actions={[
+          ...renderLeftActions(),
+          <NavToolbarSeparator leftActionsSeparator key="left-sep" />,
+          ...renderRightActions(),
+        ]}
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard/components/DashNav/DashNav.tsx:287]: I agree, can change it to this, but suspect we will need this to be array later to handle the responsive behavior of individual actions

File: public/app/core/components/AppChrome/AppChromeUpdate.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/core/components/AppChrome/AppChromeUpdate.tsx:9]: hmmmm interesting. I understand. Not sure what to think about that, apart from possibly needing a higher-level/specific API for that when we get to it.  Maybe. idk.

FWIW there's https://reactjs.org/docs/react-api.html#reactchildrentoarray and family, but not sure if useful. for this.

**Comment by torkelo** [comment]
@joshhunt pushed an update to remove/simplify things we don't need yet until we know how to handle the responsive stuff

File: public/app/core/components/AppChrome/NavToolbarActions.tsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { css } from '@emotion/css';
+import React, { PropsWithChildren } from 'react';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+
+export interface Props extends PropsWithChildren<{}> {}
+
+export function NavToolbarActions({ children }: Props) {
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/core/components/AppChrome/NavToolbarActions.tsx]: removed it for now

**Review by joshhunt — APPROVED** [review]

**Pull request merged**